graph/options/options.go

```diff
@@ -64,7 +64,10 @@ func NewOptions(r *http.Request) Options {
 	var namespaceNames []string
 	fetchNamespaces := namespaces == NamespaceAll || (namespaces == "" && (namespace == NamespaceAll))
```

Comment:
I'm not exactly sure where is the best place to put my overall review comment so I'll put it here.  First let me make sure I understand what we're trying to do.  I think what this PR wants to do is two things: first, if a caller requests an inaccessible namespace return an error.  Second, for [outsider] nodes in inaccessible namespaces, flag the node as inaccessible so that the UI can handle it correctly.  If that is a valid synopsis I think the impl can be simplified to be a lot less intrusive. 

I'd suggest the following:

options.go is responsible for the up front processing of the requested graph options.  It's a good place to determine the accessible namespaces and validate the requested namespaces.  I'd suggest always fetching the accessible namespaces.  For 'all' namespace graphs, you'd then use that list.  If the user requested explicit namespaces then do the validate here and return an error immediately if a requested namespace is inaccessible.  We already have Options.Namespaces to list the namespaces to include in the graph,  Add a new field, Options.AccessibleNamespaces that provides the full list of accessible namespaces.  If we do this the graph.go code can assume that Options.Namespaces has only accessible namespaces.  And it is provided the accessible namespaces to be used to flag outsider nodes.

Then, generate the graph normally, just like before.  No need for new parameters or queries for accessible namespaces.

Finally, after all of the graph generation is done, add some code, probably in the graph.isOutside() function, to set the isAccessible flag.  If I understand correctly, only outsider nodes could be inaccessible as we are guaranteed that the "insider" nodes are already for accessible namespaces.

Maybe I am misunderstanding, but I think if you take this approach you can eliminate a lot of the changes that added complexity to this PR, and leave the main graph generation logic untouched.